src/clarity-angular/dropdown/dropdown-submenu.ts

```diff
@@ -0,0 +1,41 @@
+import {Component, ContentChildren, Input, QueryList} from "@angular/core";
+import {DropdownItem} from "./dropdown-item";
+
+@Component({
+    selector: "clr-dropdown-submenu",
+    template: `
+        <ng-content></ng-content>
+    `,
+    host: {
+        "[class.dropdown-submenu]" : "true",
+        "[class.open]" : "open",
+        "[class.right]" : "openDirection === 'right'",
+        "[class.left]" : "openDirection === 'left'",
+        "(click)": "open = true",
+        "(mouseenter)": "open = true",
+        "(mouseleave)": "open = false"
+    }
+})
+export class DropdownSubmenu {
+    private _open: boolean = false;
+    @Input ("clrDropdownSubmenuOpenDirection") openDirection: string = "right";
+
+    @ContentChildren(DropdownItem, {descendants: false}) menuItems: QueryList<DropdownItem>;
```

Comment:
Actually, the opposite. By passing in `{descendants: false}` (it's `true` by default if you don't specify), we are only checking for the direct children. That way at each level of submenu, we validate the existence of a child element with a `clrDropdownItem` directive.